--- tracker-0.12.8/src/libtracker-data/Makefile.am.link	2011-11-26 12:16:34.974657920 +0800
+++ tracker-0.12.8/src/libtracker-data/Makefile.am	2011-11-26 12:16:57.444655161 +0800
@@ -47,6 +47,7 @@
 libtracker_data_la_LIBADD =                            \
 	$(top_builddir)/src/gvdb/libgvdb.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la \
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql.la \
 	$(BUILD_LIBS)                                  \
 	$(LIBTRACKER_DATA_LIBS)                        \
 	-lm
--- tracker-0.12.8/src/libtracker-data/Makefile.in.link	2011-11-26 12:16:42.804656308 +0800
+++ tracker-0.12.8/src/libtracker-data/Makefile.in	2011-11-26 12:17:05.705120131 +0800
@@ -516,6 +516,7 @@
 
 libtracker_data_la_LIBADD = $(top_builddir)/src/gvdb/libgvdb.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la \
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql.la \
 	$(BUILD_LIBS) $(LIBTRACKER_DATA_LIBS) -lm $(am__append_1)
 libtracker_data_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
